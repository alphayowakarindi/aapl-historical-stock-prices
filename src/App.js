import { useEffect } from 'react';
import { useDispatch } from 'react-redux';
import { Link, Route, Routes } from 'react-router-dom';
import Home from './pages/Home';
import StockDetails from './pages/StockDetails';
import { fetchStockHistory } from './redux/stockHistory/stockHistory';

function App() {
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(fetchStockHistory());
  });
  return (
    <div>
      <nav>
        <h1>AAPL</h1>
        <Link to="/">Home</Link>

      </nav>

      <Routes>
        <Route path="/" element={<Home />} />
        <Route path="stock-details/:date" element={<StockDetails />} />
      </Routes>
    </div>
  );
}

export default App;
