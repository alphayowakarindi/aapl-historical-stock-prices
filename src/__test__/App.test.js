import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../redux/configureStore';
import { BrowserRouter } from 'react-router-dom';
import App from '../App';

test('renders when the app component matches snapshot', () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Provider store={store}>
          <App />
        </Provider>
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
