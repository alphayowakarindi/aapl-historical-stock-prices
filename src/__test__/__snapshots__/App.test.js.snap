// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders when the app component matches snapshot 1`] = `
<div>
  <nav>
    <h1>
      AAPL Daily Stock Prices
    </h1>
    <div
      className="nav-svgs"
    >
      <svg
        className="w-6 h-6"
        fill="none"
        height="30px"
        stroke="currentColor"
        strokeWidth={1.5}
        viewBox="0 0 30 30"
        width="30px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9.594 3.94c.09-.542.56-.94 1.11-.94h2.593c.55 0 1.02.398 1.11.94l.213 1.281c.063.374.313.686.645.87.074.04.147.083.22.127.324.196.72.257 1.075.124l1.217-.456a1.125 1.125 0 011.37.49l1.296 2.247a1.125 1.125 0 01-.26 1.431l-1.003.827c-.293.24-.438.613-.431.992a6.759 6.759 0 010 .255c-.007.378.138.75.43.99l1.005.828c.424.35.534.954.26 1.43l-1.298 2.247a1.125 1.125 0 01-1.369.491l-1.217-.456c-.355-.133-.75-.072-1.076.124a6.57 6.57 0 01-.22.128c-.331.183-.581.495-.644.869l-.213 1.28c-.09.543-.56.941-1.11.941h-2.594c-.55 0-1.02-.398-1.11-.94l-.213-1.281c-.062-.374-.312-.686-.644-.87a6.52 6.52 0 01-.22-.127c-.325-.196-.72-.257-1.076-.124l-1.217.456a1.125 1.125 0 01-1.369-.49l-1.297-2.247a1.125 1.125 0 01.26-1.431l1.004-.827c.292-.24.437-.613.43-.992a6.932 6.932 0 010-.255c.007-.378-.138-.75-.43-.99l-1.004-.828a1.125 1.125 0 01-.26-1.43l1.297-2.247a1.125 1.125 0 011.37-.491l1.216.456c.356.133.751.072 1.076-.124.072-.044.146-.087.22-.128.332-.183.582-.495.644-.869l.214-1.281z"
          strokeLinecap="round"
          strokeLinejoin="round"
        />
        <path
          d="M15 12a3 3 0 11-6 0 3 3 0 016 0z"
          strokeLinecap="round"
          strokeLinejoin="round"
        />
      </svg>
      <svg
        className="w-6 h-6"
        fill="none"
        height="30px"
        stroke="currentColor"
        strokeWidth={1.5}
        viewBox="0 0 30 30"
        width="30px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 18.75a6 6 0 006-6v-1.5m-6 7.5a6 6 0 01-6-6v-1.5m6 7.5v3.75m-3.75 0h7.5M12 15.75a3 3 0 01-3-3V4.5a3 3 0 116 0v8.25a3 3 0 01-3 3z"
          strokeLinecap="round"
          strokeLinejoin="round"
        />
      </svg>
    </div>
  </nav>
  <div
    className="stock-container"
  />
</div>
`;
